client/hotrod-client-new/src/test/java/org/infinispan/client/hotrod/impl/iteration/BaseIterationFailOverTest.java

```diff
@@ -70,8 +70,6 @@ public void testFailOver() throws InterruptedException {
 
       }
 
-      assertEquals(cacheSize, entries.size());
```

Comment:
The operation marks the segments missing and it causes missing entries at end? Or does the operation retries the operation to succeed eventually?

Reply:
You mean how the client works? What it does is keep retrying different servers with hopefully a new topology each time until it has confirmed it has received all entries for each segment.

https://github.com/infinispan/infinispan/blob/main/client/hotrod-client-new/src/main/java/org/infinispan/client/hotrod/impl/iteration/RemotePublisher.java#L142

All of the iteration commands do no support retry. It is all handled by the publisher to do the retries.

Reply:
I removed this line on the test, because the next assertion already does this and actually reports the sets values if they don't match.